dts/bindings/counter/nxp,lptmr.yaml

```diff
@@ -34,3 +34,14 @@ properties:
     type: int
     required: true
     description: Represents the width of the timer in bits.
+
+  counter-mode-enable:
+    type: boolean
+    description: |
+      Enable Counter or Pulse Mode.
+      During Counter mode, the Clock will increment on
+      every cycle of the chosen clk-source.
+      During Pulse Mode, the clock will increment on
+      every assertion of the chosen clk-source.
```

Comment:
This description does not help me understand what is the difference, after all isn't a clock signal just a periodic assertion?

Reply:
Pulse mode it is used to count pulses coming from an external pin selected by`input-pin`,  provided the correct clock source was also selected.

Reply:
the description is not sufficient about this then, what @manuargue said here makes more sense than what is in the PR